kernel/timeout.c

```diff
@@ -326,6 +326,25 @@ k_timeout_t sys_timepoint_timeout(k_timepoint_t timepoint)
 	return K_TICKS(remaining);
 }
 
+bool sys_timepoint_lessthan(k_timepoint_t a, k_timepoint_t b)
+{
+	return a.tick < b.tick;
+}
+
+bool sys_timeout_lessthan(k_timeout_t a, k_timeout_t b)
+{
+	if (IS_ENABLED(CONFIG_TIMEOUT_64BIT)) {
+		if (K_TIMEOUT_EQ(a, K_FOREVER)) {
+			return false;
+		} else if (K_TIMEOUT_EQ(b, K_FOREVER)) {
```

Comment:
No need for `else if` after a `return`.

```suggestion
		}
		if (K_TIMEOUT_EQ(b, K_FOREVER)) {
```